tests/FsUnit.Xunit.Test/typed.shouldEqualTests.fs

```
@@ -0,0 +1,130 @@
+        1 |> shouldEqual 1
+
+    [<Fact>]
+    member __.``value type should fail to equal nonequivalent value``() =
+        shouldFail(fun () -> 1 |> shouldEqual 2)
+
+    [<Fact>]
+    member __.``value type should not equal nonequivalent value``() =
+        1 |> shouldNotEqual 2
+
+    [<Fact>]
+    member __.``value type should fail to not equal equivalent value``() =
+        shouldFail(fun () -> 1 |> shouldNotEqual 1)
+
+    [<Fact>]
+    member __.``reference type should equal itself``() =
+        anObj |> shouldEqual anObj
+
+    [<Fact>]
+    member __.``reference type should fail to equal other``() =
+        shouldFail(fun () -> anObj |> shouldEqual otherObj)
+
+    [<Fact>]
+    member __.``reference type should not equal other``() =
+        anObj |> shouldNotEqual otherObj
+
+    [<Fact>]
+    member __.``reference type should fail to not equal itself``() =
+        shouldFail(fun () -> anObj |> shouldNotEqual anObj)
+
+    [<Fact>]
+    member __.``should pass when Equals returns true``() =
+        anObj |> shouldEqual(box(new AlwaysEqual()))
+
+    [<Fact>]
+    member __.``should fail when Equals returns false``() =
+        shouldFail(fun () -> anObj |> shouldEqual(box(new NeverEqual())))
+
+    [<Fact>]
+    member __.``should pass when negated and Equals returns false``() =
+        anObj |> shouldNotEqual(box(new NeverEqual()))
+
+    [<Fact>]
+    member __.``should fail when negated and Equals returns true``() =
+        shouldFail(fun () -> anObj |> shouldNotEqual(box(new AlwaysEqual())))
+
+    [<Fact>]
+    member __.``None should equal None``() =
+        None |> shouldEqual None
+
+    [<Fact>]
+    member __.``Error "Foo" should equal Error "Foo"``() =
+        Error "Foo" |> shouldEqual(Error "Foo")
+
+    [<Fact>]
+    member __.``Error "Foo" should equal fails and have same message``() =
+        (fun () -> Error "Foo" |> shouldEqual(Error "Bar"))
+        |> Assert.Throws<System.NullReferenceException>
```

`MatchException` is thrown instead of `NullReferenceException `